stage2/06-connection-details/01-run.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash -e
+
+echo "Installing connection-details and dependencies"
+
+cp files/Pipfile* ${ROOTFS_DIR}/
```

Comment:
Quote to avoid word splitting, and `{` & `}` are pointless.

Reply:
Good point re quotes.

Re `{}` I've started using them everywhere in shell scripting, I think it's a good idea.

In this specific case it's not needed since the slash can't be confused with a var name, but there are many situations where the braces can save you from messing something up. I find it easier to just use braces everywhere than to stop and think each time I'm writing a variable if the braces are necessary or not.

It's more consistent, requires less thinking, and there are no downsides.

What do you think?

Reply:
I've been using vars with _'stache{_ since forever, but recently I've started switching.

Honestly, I don't have strong reasons.  There are some minor ones like removing clutter and noise;  having each line shorter by `2*$num_vars_used` chars, which makes it easier (for me) to scan, and parse;  better autocompletion support in IDEs;  the fact that when they are used then, they actually carry meaning, ex: permutation, or var neighbours sth that can be interpreted as part of the name, which brings me to my, I guess, main reason:

In practice, the only non-obvious thing to remember is `_`.  All else is kinda obvious `/a-z0-9_/i`: 
 
* `$minutesm` -> `${minutes}m`
* `$ROUND9` -> `${ROUND}9`
* `$NAME_sth.zip` -> `${NAME}_sth.zip`.

PS. I'm assuming you're making a special exception for `${1}`, `${?}`, `${*}`, `${@}`, etc?

PS2. Guess what the following will print :P

```shell
fn() { echo "$10 ${10}"; }; fn $(seq 4 20)
```

Guess both are fine, just maybe one optimizes more for writing and another for reading 🤷🏻‍♂️.

Reply:
PS2. Guess what the following will print :P

```shell
fn() { echo "$10 ${10}"; }; fn $(seq 4 20)
```

I guessed `13 13` which is what zsh prints but bash prints `40 13`.

Isn't this another reason in favour of braces?